armi/physics/neutronics/tests/test_crossSectionManager.py

```diff
@@ -468,25 +471,57 @@ def test_createRepresentativeBlocksUsingExistingBlocks(self):
         )
         self.assertEqual(origXSIDsFromNew["BA"], "AA")
 
-    def test_interactCoupled_UpdateTrue(self):
-        """ensure that representativeBlocks get populated if timeNode == 0"""
+    def test_interactBOL(self):
+        """Test `BOL` lattice physics update frequency"""
         self.blockList[0].r.p.timeNode = 0
-        self.csm.interactCoupled(iteration=0)
+        self.csm.cs[CONF_LATTICE_PHYSICS_UPDATE_FREQUENCY] = "BOL"
+        self.csm.interactBOL()
         self.assertTrue(self.csm.representativeBlocks)
 
-    def test_interactCoupled_UpdateFalse(self):
-        """ensure that representativeBlocks remains empty if timeNode == 1"""
+    def test_interactBOC(self):
+        """Test `BOC` lattice physics update frequency"""
+        self.blockList[0].r.p.timeNode = 0
+        self.csm.cs[CONF_LATTICE_PHYSICS_UPDATE_FREQUENCY] = "BOC"
+        self.csm.interactBOL()
+        self.csm.interactBOC()
+        self.assertTrue(self.csm.representativeBlocks)
+
+    def test_interactEveryNode(self):
+        """Test `everyNode` lattice physics update frequency"""
         self.blockList[0].r.p.timeNode = 1
-        self.csm.interactCoupled(iteration=0)
+        self.csm.cs[CONF_LATTICE_PHYSICS_UPDATE_FREQUENCY] = "BOC"
+        self.csm.interactBOL()
+        self.csm.interactEveryNode()
         self.assertFalse(self.csm.representativeBlocks)
+        self.csm.cs[CONF_LATTICE_PHYSICS_UPDATE_FREQUENCY] = "everyNode"
+        self.csm.interactBOL()
```

Comment:
So I walked this through a debugger and can see that for this test to pass the repeated calls to `interactBOL` are needed, but I don't know why. Can you explain?

Reply:
`interactBOL` sets the `self.csm._latticePhysicsFrequency`. If you just run `interactEverynode`, it doesn't update that attribute with the new setting.